DynamicTablesPkg/Include/AcpiObjects.h

```diff
@@ -139,6 +140,107 @@ typedef struct AmlPsdInfo {
   UINT32    NumProc;
 } AML_PSD_INFO;
 
+/** A structure that describes a
+    processor power state control package Info.
+
+  Cf. ACPI 6.4, s8.4.2.1 _CST (Processor Power State Control).
+  Package {
+    Register  // Buffer (Resource Descriptor)
+    Type      // Integer (BYTE)
+    Latency   // Integer (WORD)
+    Power     // Integer (DWORD)
+  }
+*/
+typedef struct AmlCstPkgInfo {
+  EFI_ACPI_6_4_GENERIC_ADDRESS_STRUCTURE    Register;
+  UINT8                                     Type;
+  UINT16                                    Latency;
+  UINT32                                    Power;
+} AML_CST_PKG_INFO;
+
+/** A structure that describes a
+    processor power state control Info.
+
+  Cf. ACPI 6.4, s8.4.1.1 _CST (Processor Power State Control).
+  Package {
+    Count             // Integer
+    CStates[0]        // Package
+    ...
+    CStates[Count-1]  // Package
+  }
+  Where Package is defined as AML_CST_PKG_INFO:
+  Package {
+    Register    // Buffer (Resource Descriptor)
+    Type        // Integer (BYTE)
+    Latency     // Integer (WORD)
+    Power       // Integer (DWORD)
+  }
+*/
+typedef struct AmlCstInfo {
+  UINT32              Count;
+  AML_CST_PKG_INFO    *CStates;

```

Comment:
Unfortunately it is not possible to use pointers to reference other objects with the DynamicTablesPkg. You will have to use `CM_OBJECT_TOKEN`.
Indeed, a module could create `AML_CST_PKG_INFO` objects, give them to the ConfigurationManager, and then be unloaded. The pointers would then either be invalid, or would have to be modified.

To avoid this kind of situation, the `AML_CST_INFO` should replace the pointer with a `CM_OBJECT_TOKEN`. This token will have to reference a `CM_ARCH_COMMON_OBJ_REF`, containing an array of tokens. This array of token will then reference the `AML_CST_PKG_INFO` objects a processor can use.

There are other places where this mechanism is used. Please search for  `CM_ARCH_COMMON_OBJ_REF` usage